R/initialization.R

```diff
@@ -4,82 +4,84 @@
+  # Define some variables
   pop <- stan_data$state_pop
-
   n_weeks <- as.numeric(stan_data$n_weeks)
   tot_weeks <- as.numeric(stan_data$tot_weeks)
   ot <- as.numeric(stan_data$ot)
   ht <- as.numeric(stan_data$ht)
-
+  n_subpops <- as.numeric(stan_data$n_subpops)
+  n_ww_lab_sites <- as.numeric(stan_data$n_ww_lab_sites)
   # Estimate of number of initial infections
-  i0 <- mean(stan_data$hosp[1:7], na.rm = TRUE) / p_hosp_mean
+  i0 <- mean(stan_data$hosp[1:7], na.rm = TRUE) / params$p_hosp_mean
 
   n_subpops <- as.numeric(stan_data$n_subpops)
   n_ww_lab_sites <- as.numeric(stan_data$n_ww_lab_sites)
 
   init_list <- list(
-    w = stats::rnorm(n_weeks - 1, 0, 0.01),
-    eta_sd = abs(stats::rnorm(1, 0, 0.01)),
-    eta_i0 = abs(stats::rnorm(n_subpops, 0, 0.01)),
-    sigma_i0 = abs(stats::rnorm(1, 0, 0.01)),
-    eta_growth = abs(stats::rnorm(n_subpops, 0, 0.01)),
-    sigma_growth = abs(stats::rnorm(1, 0, 0.01)),
+    w = stats::rnorm(n_weeks - 1, 0, stdev),
+    eta_sd = abs(stats::rnorm(1, 0, stdev)),
+    eta_i0 = abs(stats::rnorm(n_subpops, 0, stdev)),
+    sigma_i0 = abs(stats::rnorm(1, 0, stdev)),
+    eta_growth = abs(stats::rnorm(n_subpops, 0, stdev)),
+    sigma_growth = abs(stats::rnorm(1, 0, stdev)),
     autoreg_rt = abs(stats::rnorm(
       1,
-      autoreg_rt_a / (autoreg_rt_a + autoreg_rt_b),
+      params$autoreg_rt_a / (params$autoreg_rt_a + params$autoreg_rt_b),
       0.05
     )),
     log_r_mu_intercept = stats::rnorm(
       1,
-      convert_to_logmean(1, 0.1),
-      convert_to_logsd(1, 0.1)
+      convert_to_logmean(1, stdev),
+      convert_to_logsd(1, stdev)
     ),
     error_site = matrix(
       stats::rnorm(n_subpops * n_weeks,
         mean = 0,
-        sd = 0.1
+        sd = stdev
       ),
       n_subpops,
       n_weeks
     ),
     autoreg_rt_site = abs(stats::rnorm(1, 0.5, 0.05)),
     autoreg_p_hosp = abs(stats::rnorm(1, 1 / 100, 0.001)),
-    sigma_rt = abs(stats::rnorm(1, 0, 0.01)),
+    sigma_rt = abs(stats::rnorm(1, 0, stdev)),
     i0_over_n = stats::plogis(stats::rnorm(1, stats::qlogis(i0 / pop), 0.05)),
-    initial_growth = stats::rnorm(1, 0, 0.001),
+    initial_growth = stats::rnorm(1, 0, stdev),
     inv_sqrt_phi_h = 1 / sqrt(200) + stats::rnorm(1, 1 / 10000, 1 / 10000),
     sigma_ww_site_mean = abs(stats::rnorm(
-      1, sigma_ww_site_prior_mean_mean,
-      0.1 * sigma_ww_site_prior_mean_sd
+      1, params$sigma_ww_site_prior_mean_mean,
+      stdev * params$sigma_ww_site_prior_mean_sd
     )),
     sigma_ww_site_sd = abs(stats::rnorm(
-      1, sigma_ww_site_prior_sd_mean,
-      0.1 * sigma_ww_site_prior_sd_sd
+      1, params$sigma_ww_site_prior_sd_mean,
+      stdev * params$sigma_ww_site_prior_sd_sd
     )),
-    sigma_ww_site_raw = abs(stats::rnorm(n_ww_lab_sites, 0, 0.05)),
-    p_hosp_mean = stats::rnorm(1, stats::qlogis(p_hosp_mean), 0.01),
-    p_hosp_w = stats::rnorm(tot_weeks, 0, 0.01),
+    sigma_ww_site_raw = abs(stats::rnorm(n_ww_lab_sites, 0, stdev)),
```

Comment:
This replacement will change the outcomes of the code; was this your intent?

Reply:
Yes, @kaitejohnson intended to change this. Longer term I would like to refactor the init simulation a bit but that's a separate PR.

Reply:
Yep I tried to do some scope creep and make this a bit more generic per #20 . Need to do more of a refactor where we sample from some constrained space centered around the center of the prior across all parameters, but for now this was my attempt at not hard coding all of the inits...